app/controllers/admin/edit_histories_controller.rb

```diff
@@ -0,0 +1,107 @@
+module Admin
+  class EditHistoriesController < ::ApplicationController
+    before_action :require_can_audit_users! #TODO other parts?
+    before_action :set_user
+
+    helper_method :name_of_whodunnit
+    helper_method :describe_changes_to
+
+    WHITELIST = %w{first_name last_name email phone agency receive_file_upload_notifications
+      notify_of_vispdat_completed notify_on_anomaly_identified}
+
+    def show
+      pt_a = PaperTrail::Version.arel_table
+      edits = PaperTrail::Version.where(pt_a[:item_id].eq(@user_id).and(pt_a[:item_type].eq('User')).
+          or(pt_a[:referenced_user_id].eq(@user_id)))
+      @versions = edits.where.not(whodunnit: nil).order(created_at: :desc).page(params[:page]).per(25)
+    end
+
+    def name_of_whodunnit(version)
+      who = version.whodunnit
+      if who.present?
+        User.find(who).name
+      end
+    end
+
+    def describe_changes_to(version)
+      item_type = version.item_type
+      results = []
+      if item_type == 'GrdaWarehouse::UserViewableEntity'
```

Comment:
move these into the models (case statement that throws an error if called on an unknown class)